service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/CarePlanMapper.java

```diff
@@ -0,0 +1,114 @@
+package uk.nhs.adaptors.oneoneone.cda.report.mapper;
+
+import lombok.AllArgsConstructor;
+import org.hl7.fhir.dstu3.model.CarePlan;
+import org.hl7.fhir.dstu3.model.Encounter;
+import org.hl7.fhir.dstu3.model.Narrative;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.springframework.stereotype.Component;
+import uk.nhs.adaptors.oneoneone.cda.report.util.NodeUtil;
+import uk.nhs.connect.iucds.cda.ucr.CE;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01ClinicalDocument1;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Component2;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Component3;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Component5;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Section;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01StructuredBody;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.hl7.fhir.dstu3.model.CarePlan.CarePlanIntent.PLAN;
+import static org.hl7.fhir.dstu3.model.CarePlan.CarePlanStatus.COMPLETED;
+import static org.hl7.fhir.dstu3.model.IdType.newRandomUuid;
+
+@Component
+@AllArgsConstructor
+public class CarePlanMapper {
+    private final String SNOMED = "2.16.840.1.113883.2.1.3.2.4.15";
+    private final String INFORMATION_ADVICE_GIVEN = "1052951000000105";
+
+    public List<CarePlan> mapCarePlan(POCDMT000002UK01ClinicalDocument1 clinicalDocument,
+                                      Encounter encounter) {
+
+        POCDMT000002UK01StructuredBody structuredBody = getStructuredBody(clinicalDocument);
+
+        return Arrays.stream(structuredBody.getComponentArray())
```

Comment:
this can throw null pointer exception?

Reply:
Why? At most you get empty list

Reply:
structureBody could be null

Reply:
Checked isSetStructuredBody first to prevent null but I this every mapper before would fail if this was ever null and I am sure the integration test would fail as the XML subbmitted would be invalid